fip-0039.md

```diff
@@ -0,0 +1,155 @@
+|Fee exceeds maximum|Actual fee is greater than supplied max_fee|400|"max_fee"|Value sent in, e.g. "1000000000"|"Fee exceeds supplied maximum"|
+|Insufficient balance|Available (unlocked and unstaked) balance in account is less than chain fee|400|"max_fee"|Value sent in, e.g. "100000000000"|"Insufficient balance"|
+|Invalid TPID|tpid format is not valid|400|"tpid"|Value sent in, e.g. "notvalidfioaddress"|"TPID must be empty or valid FIO address"|
+|Not owner of FIO Address|The signer does not own the FIO Address|403|||Type: invalid_signature|
+|Signer not actor|Signer not actor|403|||Type: invalid_signature|
+#### Response body
+|Parameter|Format|Definition|
+|---|---|---|
+|status|String|OK if successful|
+|fee_collected|Int|Amount of SUFs collected as fee|
+##### Example
+```
+{
+  "status": "OK",
+  "fee_collected": 0
+}
+```
+
+## New getters
+### Get encryption key
+Returns the FIO Public Key to use for encryption of New Funds Request/Record OBT content blob.
+#### New end point: */get_encrypt_key* 
+#### Request body
+|Parameter|Required|Format|Definition|
+|---|---|---|---|
+|fio_address|Yes|FIO Public Key|FIO Address to fetch encrypt key for.|
+##### Example
+```
+{
+  "fio_address": "purse@alice"
+}
+```
+##### Processing
+* Request is validated per Exception handling
+* Return encrypt_public_key set using *updcryptkey*. If not set return FIO Public Key mapped using add_pub_address for chain_code: FIO, token_code: FIO
+##### Exception handling
+|Error condition|Trigger|Type|fields:name|fields:value|Error message|
+|---|---|---|---|---|---|
+|Invalid FIO Address|Format of FIO Address not valid or FIO Address does not exist.|400|"fio_address"|Value sent in, i.e. "purse@alice"|"FIO Address invalid or does not exist."|
+#### Response body
+|Parameter|Format|Definition|
+|---|---|---|
+|encrypt_public_key|Int|FIO Public key to use for encryption.|
+##### Example
+```
+{
+  "encrypt_public_key": "FIO5G2tfhPZtiaiBU6o9SpT2XKbb28gtKUeXXarvBjxuU29VYgYCs"
+}
+```
+
+# Rationale
+The following alternative approaches were considered:
+* Enabling of _newaccount_ action. It is a system action now, but could have been made public. However, that approach was deemed more complex as it would require updates to other smart contracts which are calling this action now.
+* Storing the encryption key as a new _token_code_ for _chain_code:FIO_ using _add_pub_address_. However, it would have been harder to support backwards compatibility where the original key is returned when a user has not specified a new encryption key.
+
+# Implementation
+TBD
+
+# Backwards Compatibility
+It is recommended that _/get_encrypt_key_ is rolled out ahead of _updcryptkey_ and an advisory is issued to wallets and exchanges instructing them to start using the new getter. The current encryption key will be returned by _/get_encrypt_key_ even if unique key is not specified using _updcryptkey_.
```

Comment:
We should clarify what the process is for cases where old FIO Requests and OBT data has been encrypted using the original key. Won't this make a large amount of OBT data on the chain inaccessible?

Is this use case more for general users, or is it for power users? If it is for power users, then it may make sense to keep the default key the original mapped FIO Pub key and just let the power users create and access the "encryption pub key".